pubsub/pubsub.go

```diff
@@ -29,9 +29,36 @@ type PubSub interface {
 	Close() error
 }
 
+// BulkPublisher is the interface defining BulkPublish definition for message buses
+type BulkPublisher interface {
+	// BulkPublish is the method to publish multiple messages to a topic in a bulk fashion.
+	BulkPublish(req *BulkPublishRequest) (BulkPublishResponse, error)
+}
+
+// BulkSubscriber is the interface defining BulkSubscribe definition for message buses
+type BulkSubscriber interface {
+	// BulkSubscribe can be used to subscribe to a topic and receive messages in a bulk fashion.
+	// It will depend on broker if they can support bulk consumption.
+	// If the broker does not support bulk consumption, default implementation can be leveraged
+	// to optimize between Dapr sidecar and consumer App.
```

Comment:
- This is an implementation detail that should not be here (interface is agnostic).
- `BulkSubscribe` docs to be a part of `BulkSubscriber` (similar to above)